lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -166,6 +176,96 @@ cl_kernel OpenCLFunction::createKernel(const std::string &name,
   return kernel;
 }
 
+std::string OpenCLFunction::deviceProgramCacheDir(cl_device_id deviceId) {
+  // Support only a single device and device program cache directory
+  // for now.
+  return clDeviceProgramCacheDir;
+}
+
+std::string
+OpenCLFunction::diskCacheProgramFileName(cl_device_id deviceId,
+                                         const std::string &source,
+                                         const std::string &options) {
+  std::ostringstream hashString;
+  hashString << source << options;
+  return std::to_string(std::hash<std::string>{}(hashString.str())) + ".clb";
+}
+
+cl_program OpenCLFunction::loadProgramFromDiskCache(std::string cacheDirectory,
+                                                    std::string programFileName,
+                                                    cl_context ctx,
+                                                    cl_device_id device) {
+  std::string programPath = cacheDirectory + '/' + programFileName;
+  if (!llvm::sys::fs::exists(programPath))
+    return nullptr;
+
+  uint64_t binarySize;
+  std::error_code errc = llvm::sys::fs::file_size(programPath, binarySize);
+  CHECK(!errc) << "Error getting the file size of " << programPath << ".";
+
+  std::ifstream binFile(programPath.c_str(), std::ios::binary);
+  CHECK(binFile) << "Error opening " << programPath << " for reading.";
+
+  auto binary = glow::make_unique<unsigned char[]>(binarySize);
+  binFile.read((char *)binary.get(), binarySize);
+  CHECK(binFile) << "Could not read the binary.";
+  binFile.close();
+
+  cl_int status;
+  cl_int err;
+  cl_program prog =
+      clCreateProgramWithBinary(ctx, 1, &device, &binarySize,
```

Comment:
@pjaaskel OpenCL standard requires the fourth parameter (i.e. binarySize) go be `size_t`. This code does not compile on MacOS because of this. But `llvm::sys::fs::file_size(programPath, binarySize);` requires its last parameter to be `uint64_t`. Please fix it.